src/widgetastic/ouia.py

```diff
@@ -0,0 +1,72 @@
+from widgetastic.utils import ParametrizedLocator
+from widgetastic.xpath import quote
+from widgetastic.widget.base import Widget
+from widgetastic.widget.base import View
+from widgetastic.widget.base import ClickableMixin
+
+
+class OUIABase:
+    """
+    Base class for ``OUIA`` support. According to the spec ``OUIA`` compatible components may have
+    the following attributes in the root level HTML element:
+        * data-ouia-component-type
+        * data-ouia-component-id
+        * data-ouia-safe
+    https://ouia.readthedocs.io/en/latest/README.html#ouia-component
+    """
+
+    ROOT = ParametrizedLocator(".//*[@data-ouia-component-type={@component_type}{@component_id}]")
+
+    def set_ouia_attrs(self, component_type, component_id=None, namespace=None):
```

Comment:
this should use a leading underscore or be migrated into a cooperative mixing ctor